compiler/linkname.go

```diff
@@ -55,17 +62,32 @@ func newSymName(o types.Object) SymName {
 
 func (n SymName) String() string { return n.PkgPath + "." + n.Name }
 
+func (n SymName) IsMethod() (recv string, method string, ok bool) {
+	pos := strings.IndexByte(n.Name, '.')
+	if pos == -1 {
+		return
+	}
+	recv, method, ok = n.Name[:pos], n.Name[pos+1:], true
+	size := len(recv)
+	if size > 2 && recv[0] == '(' && recv[size-1] == ')' {
+		recv = recv[1 : size-1]
+	}
+	return
+}
+
 // parseGoLinknames processed comments in a source file and extracts //go:linkname
 // compiler directive from the comments.
 //
 // The following directive format is supported:
 // //go:linkname <localname> <importpath>.<name>
+// //go:linkname <localname> <importpath>.<type>.<name>
+// //go:linkname <localname> <importpath>.<(*type)>.<name>
```

Comment:
Could you also update documentation in doc/pargma.md? Specifically:

 - That methods are now supported too.
 - That receiver type will be converted if necessary, but not the argument types.